crates/sui-graphql-rpc/tests/dot_move_e2e.rs

```diff
@@ -0,0 +1,441 @@
+            ConnectionConfig::ci_integration_test_cfg(),
+            None,
+        )
+        .await;
+
+        let external_network_chain_id = network_cluster
+            .validator_fullnode_handle
+            .fullnode_handle
+            .sui_client
+            .read_api()
+            .get_chain_identifier()
+            .await
+            .unwrap();
+
+        println!("External chain id: {:?}", external_network_chain_id);
+
+        // // publish the dot move package in the internal resolution cluster.
+        let (pkg_id, registry_id) = publish_dot_move_package(&network_cluster).await;
+
+        let (v1, v2, v3) = publish_demo_pkg(&network_cluster).await;
+
+        let name = "app@org".to_string();
+
+        // Register the package: First, for the "base" chain state.
+        register_pkg(
+            &network_cluster,
+            pkg_id,
+            registry_id,
+            v1.clone(),
+            name.clone(),
+            None,
+        )
+        .await;
+
+        // Register the package for the external resolver.
+        register_pkg(
+            &network_cluster,
+            pkg_id,
+            registry_id,
+            v1,
+            name.clone(),
+            Some(external_network_chain_id.clone()),
+        )
+        .await;
+
+        // Initialize the internal and external clients of GraphQL.
+
+        // The first cluster uses internal resolution (mimics mainnet, does not rely on external chain).
+        let internal_client = init_dot_move_gql(
+            8000,
+            9184,
+            ServiceConfig::dot_move_test_defaults(
+                false,
+                None,
+                Some(pkg_id.into()),
+                Some(registry_id.0),
+                None,
+            ),
+        )
+        .await;
+
+        // Wait for the transactions to be committed and indexed
+        sleep(Duration::from_secs(5)).await;
+
+        // We craft a big query, which we'll use to test both the internal and the external resolution.
+        // Same query is used across both nodes, since we're testing on top of the same data, just with a different
+        // lookup approach.
+        let query = format!(
+            r#"{{ valid_latest: {}, v1: {}, v2: {}, v3: {}, v4: {} }}"#,
+            name_query(&name),
+            name_query(&format!("{}{}", &name, "/v1")),
+            name_query(&format!("{}{}", &name, "/v2")),
+            name_query(&format!("{}{}", &name, "/v3")),
+            name_query(&format!("{}{}", &name, "/v4")),
+        );
+
+        let internal_resolution = internal_client
+            .execute(query.clone(), vec![])
+            .await
+            .unwrap();
+
+        test_results(internal_resolution, &v1, &v2, &v3);
+
+        println!("Tests have finished successfully now!");
+    }
+
+    fn test_results(query_result: serde_json::Value, v1: &ObjectID, v2: &ObjectID, v3: &ObjectID) {
```

Comment:
Better to just duplicate these -- otherwise when an assertion fails, you won't have enough information to know which invocation of `test_results` failed.

Reply:
I ended up passing down an indicator and eprintln it, to avoid duplicating that.